src/back/hlsl/writer.rs

```diff
@@ -661,19 +661,49 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
 
         if global.space == crate::AddressSpace::Uniform {
             write!(self.out, " {{ ")?;
-            // Even though Naga IR matrices are column-major, we must describe
-            // matrices passed from the CPU as being in row-major order.  See
-            // the module-level comments for details.
-            if let TypeInner::Matrix { .. } = module.types[global.ty].inner {
-                write!(self.out, "row_major ")?;
-            }
-            self.write_type(module, global.ty)?;
-            let sub_name = &self.names[&NameKey::GlobalVariable(handle)];
-            write!(self.out, " {}", sub_name)?;
-            // need to write the array size if the type was emitted with `write_type`
-            if let TypeInner::Array { base, size, .. } = module.types[global.ty].inner {
-                self.write_array_size(module, base, size)?;
+
+            // We treat matrices of the form `matCx2` as a sequence of C `vec2`s
+            // (see top level module docs for details).
+            if let TypeInner::Matrix {
+                rows: crate::VectorSize::Bi,
+                columns,
+                width,
+            } = module.types[global.ty].inner
```

Comment:
This doesn't cover the case where you have:

```
@group(0)
@binding(0)
var<uniform> transform: array<mat4x2<f32>, 5>;
```

In the current code, and with this PR (I think), this generates:

```
cbuffer transform : register(b0) { float4x2 transform[5]; }
```

which is another instance of #1837.